### PR TITLE
Lockups derived key spending limit implementation.

### PR DESCRIPTION
**Pull request by poolcoke**

### CODE REVIEW

**Comment by poolcoke** [comment]
Current dependencies on/for this PR:
* feature/coin-lockups
  * **PR #678** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/678" target="_blank">Graphite</a> 
    * **PR #679** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/679" target="_blank">Graphite</a> 
      * **PR #680** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/680" target="_blank">Graphite</a> 
        * **PR #681** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/681" target="_blank">Graphite</a> 
          * **PR #682** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/682" target="_blank">Graphite</a> 
            * **PR #683** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/683" target="_blank">Graphite</a> 
              * **PR #684** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/684" target="_blank">Graphite</a> 
                * **PR #716** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/716" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/716?utm_source=stack-comment).

**Review by lazynina — CHANGES_REQUESTED** [review]
Please update `CheckIfValidUnlimitedSpendingLimit` in `network.go` and update the `AssertDependencyStructFieldNumbers` to check for 14 fields.

**Comment by poolcoke** [comment]
Fixed

**Review by lazynina — CHANGES_REQUESTED** [review]

File: lib/block_view.go [hunk 1]
```diff
@@ -2433,6 +2433,40 @@ func (bav *UtxoView) _checkAndUpdateDerivedKeySpendingLimit(
...
+	case TxnTypeCoinLockup:
+		txnMeta := txn.TxnMeta.(*CoinLockupMetadata)
+		if derivedKeyEntry, err = bav._checkLockupTxnSpendingLimitAndUpdateDerivedKey(
+			derivedKeyEntry, txnMeta.ProfilePublicKey, CoinLockupOperation); err != nil {
+			return utxoOpsForTxn, err
+		}
+	case TxnTypeUpdateCoinLockupParams:
+		txnMeta := txn.TxnMeta.(*UpdateCoinLockupParamsMetadata)
+		// Check if we're updating the transactor's yield curve.
+		if txnMeta.LockupYieldDurationNanoSecs > 0 {
```
[the remark this answers is not in the document]
Comment (poolcoke) [reply lib/block_view.go:2454]: Yeah good point. Could be some sort of an exploit where a balance is drained via fees.

**Pull request closed without merging**